scripts/apps/search/directives/SearchContainer.ts

```diff
@@ -28,6 +36,18 @@ export function SearchContainer() {
                         }
                     }
                 });
+                preferencesService.get().then((result) => {
+                    if (result !== undefined && FILTERS_PANEL_PREFERENCES_KEY in result) {
+                        this.flags.facets = result[FILTERS_PANEL_PREFERENCES_KEY]['enabled'];
+                    } else {
+                        this.flags.facets = filtersPanelPreferences[FILTERS_PANEL_PREFERENCES_KEY]['default'];
+                    }
```

Comment:
I think it'd be simpler if the preference was a boolean instead of an object.

`this.flags.facets = result[FILTERS_PANEL_PREFERENCES_KEY] ?? true`

Reply:
In this case, we can not a default value, but the main reason is to keep it consistent with other boolean preferences definitions e.q
```
superdesk.register_default_user_preference('spellchecker:status', {
        'type': 'bool',
        'enabled': True,
        'default': True
    })
```

Reply:
I think we can change it to be just a boolean, we don't really use `default` value defined on the server side, and I believe that `'type': 'bool'` is only for UI, but this setting is hidden from UI, so we are fine.

Reply:
I've tried, but server-side does not support a privilege to be just a bool.

Reply:
You could register a general preference for `global_search`
`superdesk.register_default_user_preference('global_search', {})`
and then set `filtering_open: true` inside that object.

Reply:
I've got your point, that makes sense but currently, it's designed in a different way.
Instead of having `global_search. filtering_open` object we have `search:filters_panel_open` preference key. 
With `superdesk.register_default_user_preference('global_search', {})` approach, for example, would be difficult to hide some `global_search` configs from the UI and display the rest. Also, I'm not sure how to extend it in the future.

With current notation, you can also define `category`, `privileges`, and `label`  which are used in the UI.
For example, we can have the next preferences:

```
superdesk.register_default_user_preference('search:filters_panel_open', {
        'type': 'bool',
        'enabled': True,
        'default': True
    })
superdesk.register_default_user_preference('search:some_ui_config', {
        'type': 'bool',
        'enabled': False,
        'default': False,
        'label': _('Enable some feature'),
        'category': _('Search'),
        'privileges': ['some_privelege']
    })
superdesk.register_default_user_preference('search:another_ui_config', {
        'type': 'string',
        ..
        'label': _('Another search feature'),
        'category': _('Search')
    })
```
in this case, the second and third preferences are grouped (category) and displayed in UI, first one is hidden.

Reply:
Ok, let's keep it then